ts/private/npm_repositories.bzl

```diff
@@ -28,9 +28,6 @@ def _http_archive_version_impl(rctx):
             ts = p["dependencies"]["typescript"]
         else:
             fail("key 'typescript' not found in either dependencies or devDependencies of %s" % json_path)
-        if any([not seg.isdigit() for seg in ts.split(".")]):
```

Comment:
What error message does a user get instead, if they use `ts_version_from="//package.json"` and the package.json has `"typescript": "^4.9.0"`? I suspect it will be a lot less helpful than this one

Reply:
https://github.com/aspect-build/rules_ts/blob/1a42ef4fc44a90c03d42dd5bab7e03f68a5bd739/ts/private/npm_repositories.bzl#L40-L42

I think that is the correct location for the `fail` but we can update the message...

Reply:
yeah update to "If so, you must X or Y"